game/theater/iadsnetwork/iadsnetwork.py

```diff
@@ -202,64 +219,94 @@ def initialize_basic_iads(self) -> None:
                 # available. Therefore the TGO will not get populated at all
-                logging.warning(
-                    f"IADS: No ground object found for {element_name}. This can be normal behaviour."
-                )
+
+                logging.warning(warning_msg)
                 continue
 
             # Find all connected ground_objects
             for node_name in connections:
                 try:
-                    node.add_connection_for_tgo(self.ground_objects[node_name])
-                except (KeyError):
+                    tgo = self.ground_objects[node_name]
+                    if tgo is None:
+                        raise IadsNetworkException(
+                            f"IADS: ground_objects contains None values:\n{tgo}"
+                        )
+                except KeyError:
                     logging.error(
                         f"IADS: No ground object found for connection {node_name}"
                     )
                     continue
+                node.add_connection_for_tgo(tgo)
 
     def initialize_network_from_range(self) -> None:
         """Initialize the IADS Network by range"""
         for go in self.ground_objects.values():
-            if (
-                isinstance(go, IadsGroundObject)
-                or isinstance(go, NavalGroundObject)
-                or (
-                    isinstance(go, IadsBuildingGroundObject)
-                    and IadsRole.for_category(go.category) == IadsRole.COMMAND_CENTER
-                )
-            ):
-                try:
-                    # Set as primary node
-                    node = self.node_for_tgo(go)
-                except IadsNetworkException:
+            is_iads_go = isinstance(go, IadsGroundObject)
+            is_iads_sea = isinstance(go, NavalGroundObject)
+            is_iads_cc = isinstance(go, IadsBuildingGroundObject)
+            is_iads_cc &= IadsRole.for_category(go.category) == IadsRole.COMMAND_CENTER
+            if is_iads_go or is_iads_sea or is_iads_cc:
+                # Set as primary node
+                node = self.node_for_tgo(go)
+                if node is None:
                     # TGO does not participate to iads network
                     continue
-                # Find nearby Power or Connection
-                for nearby_go in self.ground_objects.values():
-                    if nearby_go == go:
-                        continue
-                    if (
-                        IadsRole.for_category(go.category)
-                        in [
-                            IadsRole.POWER_SOURCE,
-                            IadsRole.CONNECTION_NODE,
-                        ]
-                        and nearby_go.position.distance_to_point(go.position)
-                        <= node.group.iads_role.connection_range.meters
-                    ):
-                        node.add_connection_for_tgo(nearby_go)
```

Comment:
Refactor this into `_connect_to_network` since the logic is sound and I need to reuse this...